server/src/main/java/org/elasticsearch/index/reindex/UpdateByQueryRequest.java

```diff
@@ -59,8 +64,91 @@ private UpdateByQueryRequest(SearchRequest search, boolean setDefaults) {
     /**
      * Set the ingest pipeline to set on index requests made by this action.
      */
-    public void setPipeline(String pipeline) {
+    public UpdateByQueryRequest setPipeline(String pipeline) {
         this.pipeline = pipeline;
+        return this;
+    }
+
+    /**
+     * Set the query for selective update
+     */
+    public UpdateByQueryRequest setQuery(QueryBuilder query) {
+        if (query != null) {
+            getSearchRequest().source().query(query);
+        }
+        return this;
+    }
+
+    /**
+     * Set the document types for the update
+     */
+    public UpdateByQueryRequest setDocTypes(String... types) {
+        if (types != null) {
+            getSearchRequest().types(types);
+        }
+        return this;
+    }
+
+    /**
+     * Set the indices on which the update by query request is to be run
+     */
+    public UpdateByQueryRequest setIndices(String... indices) {
```

Comment:
This method is not required. Usage should be replaced by the method `indices(String... indices)`.